inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/CommonConfigHolder.java

```diff
@@ -0,0 +1,477 @@
+        if (StringUtils.isNotEmpty(tmpValue)) {
+            this.responseAfterSave = "TRUE".equalsIgnoreCase(tmpValue.trim());
+        }
+        // read max response after save timeout
+        tmpValue = this.props.get(KEY_MAX_RAS_TIMEOUT_MS);
+        if (StringUtils.isNotEmpty(tmpValue)) {
+            this.maxResAfterSaveTimeout = NumberUtils.toLong(tmpValue.trim(), VAL_DEF_MAX_RAS_TIMEOUT_MS);
+        }
+        // read max bufferqueue size
+        tmpValue = this.props.get(KEY_MAX_BUFFERQUEUE_SIZE_KB);
+        if (StringUtils.isNotEmpty(tmpValue)) {
+            this.maxBufferQueueSizeKb = NumberUtils.toInt(tmpValue.trim(), VAL_DEF_MAX_BUFFERQUEUE_SIZE_KB);
+        }
+        // read event handler
+        tmpValue = this.props.get(KEY_EVENT_HANDLER);
+        if (StringUtils.isNotBlank(tmpValue)) {
+            this.eventHandler = tmpValue.trim();
+        }
+        // read cache cluster selector
+        tmpValue = this.props.get(KEY_CACHE_CLUSTER_SELECTOR);
+        if (StringUtils.isNotBlank(tmpValue)) {
+            this.cacheClusterSelector = tmpValue.trim();
+        }
+        // read proxy node id
+        tmpValue = this.props.get(KEY_PROXY_NODE_ID);
+        if (StringUtils.isNotBlank(tmpValue)) {
+            this.proxyNodeId = tmpValue.trim();
+        }
+        // read msg compress type
+        tmpValue = this.props.get(KEY_MSG_SENT_COMPRESS_TYPE);
+        if (StringUtils.isNotBlank(tmpValue)) {
+            this.msgCompressType = tmpValue.trim();
+        }
+        // read prometheus Http Port
+        tmpValue = this.props.get(KEY_PROMETHEUS_HTTP_PORT);
+        if (StringUtils.isNotEmpty(tmpValue)) {
+            this.prometheusHttpPort = NumberUtils.toInt(tmpValue.trim(), VAL_DEF_PROMETHEUS_HTTP_PORT);
+        }
+        // initial ip parser
+        try {
+            Class<? extends IManagerIpListParser> ipListParserClass =
+                    (Class<? extends IManagerIpListParser>) Class.forName(this.managerType);
+            this.ipListParser = ipListParserClass.getDeclaredConstructor().newInstance();
+            this.ipListParser.setCommonProperties(this.props);
+        } catch (Throwable t) {
+            LOG.error("Initial ipListParser Class {} failure, exit!", this.managerType, t);
+            System.exit(6);
+        }
+    }
+
+    private void chkRequiredFields(String requiredFieldKey) {
+        String fieldVal = props.get(requiredFieldKey);
+        if (fieldVal == null) {
+            LOG.error("Missing mandatory field {} in {}, exit!",
+                    requiredFieldKey, COMMON_CONFIG_FILE_NAME);
+            System.exit(4);
+        }
+        if (StringUtils.isBlank(fieldVal)) {
+            LOG.error("Required {} field value is blank in {}, exit!",
+                    requiredFieldKey, COMMON_CONFIG_FILE_NAME);
+            System.exit(5);
+        }
+    }
+
+    private boolean loadConfigFile() {
+        InputStream inStream = null;
+        try {
```

Comment:
Using `try-with-resource` can reduce the `finally` statement.

Reply:
This is just a coding habit and not a problem. Try-with-resource is just a syntactic sugar. Personally, I tend to explicitly clean up resources during the release phase.